hibernate-core/src/main/antlr/hql-sql.g

```diff
@@ -436,7 +436,7 @@ selectExpr
 	;
 
 count
-    : #(COUNT  { inCount = true; } ( DISTINCT { inCountDistinct = true; } | ALL )? ( aggregateExpr | ROW_STAR ) ) {
+    : #(COUNT  { inCount = true; } ( DISTINCT { inCountDistinct = true; } | ALL )? ( aggregateExpr | ROW_STAR | query) ) {
```

Comment:
But the JIRA issue shows a different query:

`select count(distinct generatedAlias0.claimNumber || generatedAlias1.cmsHcpcsCodeId) from harbinger.sdk.data.ClaimTransaction as generatedAlias1 left join generatedAlias1.claim as generatedAlias0`

I'd expect such a query in the test case.